### PR TITLE
Await trashbin deletion requests

### PR DESCRIPTION
**Pull request by skshetry**
## Description

Await deletion requests.

## Related Issue

- Fixes #2091 

## How Has This Been Tested?

- locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/5286/

<p>






























</p></details>

File: tests/acceptance/helpers/webdavHelper.js [hunk 1]
```diff
@@ -74,7 +73,7 @@ exports.move = function (userId, fromName, toName) {
  * @param {array} properties
  * @param {number} folderDepth
  */
-exports.propfind = function (path, userId, password, properties, folderDepth = 1) {
```
Comment (individual-it) [inline tests/acceptance/helpers/webdavHelper.js]: put this kind of changes in a separate PR and adjust the function docs

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUILogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/5322/

<p>





</p></details>

**Comment by skshetry** [comment]
Rebased, as #2109 got merged. Hopefully, `webUILogin` will now pass.

File: tests/acceptance/stepDefinitions/filesContext.js [hunk 2]
```diff
@@ -121,10 +121,15 @@ When('the user deletes the following elements using the webUI', function (table)
   return client.page.filesPage()
 })
 
-Given('the following files have been deleted by user {string}', function (user, table) {
-  for (const line of table.rows()) {
-    webdav.delete(user, line[0])
-  }
+Given('the following files have been deleted by user {string}', async function (user, table) {
+  const filesToDelete = table.rows()
+  await Promise.all(
```
Comment (skshetry) [inline tests/acceptance/stepDefinitions/filesContext.js]: As we send multiple requests at once, this might fail for some stupid scenarios we want to test. 
Eg:
```feature
Given the following files have been deleted by user "user0"
 | simple-folder           |
 | simple-folder/lorem.txt |
```
As the folder will be locked, anything might happen. `lorem.txt` might get deleted first, and then the folder itself, and `simple-folder` might get deleted and get locked.
